x/auth/vesting/client/cli/cli_test.go

```diff
@@ -0,0 +1,158 @@
+package cli_test
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/gogo/protobuf/proto"
+
+	"github.com/stretchr/testify/suite"
+
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	"github.com/cosmos/cosmos-sdk/crypto/hd"
+	"github.com/cosmos/cosmos-sdk/crypto/keyring"
+	"github.com/cosmos/cosmos-sdk/testutil"
+	clitestutil "github.com/cosmos/cosmos-sdk/testutil/cli"
+
+	"github.com/cosmos/cosmos-sdk/testutil/network"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/auth/vesting/client/cli"
+	banktestutil "github.com/cosmos/cosmos-sdk/x/bank/client/testutil"
+)
+
+type IntegrationTestSuite struct {
+	suite.Suite
+
+	cfg     network.Config
+	network *network.Network
+}
+
+func (s *IntegrationTestSuite) SetupSuite() {
+	s.T().Log("setting up integration test suite")
+
+	s.cfg = network.DefaultConfig()
+
+
+	s.network = network.New(s.T(), s.cfg)
+
+	_, err := s.network.WaitForHeight(1)
+	s.Require().NoError(err)
+
+	_, err = s.network.WaitForHeight(1)
+	s.Require().NoError(err)
+}
+
+func (s *IntegrationTestSuite) TestCreateClawbackVestingAccountCmd() {
+	val := s.network.Validators[0]
+
+	info, _, err := val.ClientCtx.Keyring.NewMnemonic("NewCreatePoolAddr",
+		keyring.English, sdk.FullFundraiserPath, keyring.DefaultBIP39Passphrase, hd.Secp256k1)
+	s.Require().NoError(err)
+	newAddr := sdk.AccAddress(info.GetPubKey().Address())
+	_, err = banktestutil.MsgSendExec(
+		val.ClientCtx,
+		val.Address,
+		newAddr,
+		sdk.NewCoins(sdk.NewInt64Coin(s.cfg.BondDenom, 200000000), sdk.NewInt64Coin("node0token", 20000)), fmt.Sprintf("--%s=true", flags.FlagSkipConfirmation),
+		fmt.Sprintf("--%s=%s", flags.FlagBroadcastMode, flags.BroadcastBlock),
+	)
+	s.Require().NoError(err)
+
+	testCases := []struct {
+		name         string
+		lockupJson   string
+		vestingJson  string
+		expectErr    bool
+		respType     proto.Message
+		expectedCode uint32
+	}{
+		{
+			"create vesting account",
```

Comment:
Agreed that those the test cases that you've mentioned are good edge cases! 

Do you think they should go under `cli_test.go`, or `vesting_account_test.go`? imo, I don't think it differs on where the test would go under

Reply:
I think that, ideally, having tests at all levels of abstraction is necessary.

Let's consider 2 examples:
1. We only have edge case tests in `cli_test.go`. A test failure occurs in a few months after a this change. Then, the engineer working on fixing the bug cannot immediately tell where exactly the bug happened. Was it in `cli_test.go` or `vesting_account_test.go`?
2. We only have tests in `vesting_account_test.go`. In a few months, someone who didn't work on this PR introduced a change on the CLI level and makes a logic error that never gets caught because we never had edge case tests covering the cli level of abstraction